Osiris/Hacks/Backtrack.cpp

```diff
@@ -0,0 +1,168 @@
+
+    if (stage == FrameStage::NET_UPDATE_END) {
+        const auto localPlayer = interfaces.entityList->getEntity(interfaces.engine->getLocalPlayer());
+        if (!localPlayer)
+            return;
+
+        for (int i = 1; i <= interfaces.engine->getMaxClients(); i++) {
+            auto entity = interfaces.entityList->getEntity(i);
+            if (!entity || entity == localPlayer || entity->isDormant() || !entity->isAlive()
+                || !entity->isEnemy())
+            {
+                if (records[i].size() > 2)
+                    records[i].clear();
+
+                continue;
+            }
+
+            auto& rec = records[i];
+            for (auto r = rec.begin(); r != rec.end();) {
+                if (!valid(r->simulationTime))
+                    r = rec.erase(r);
+                else
+                    r++;
+            }
+
+            if (records[i].size() && (records[i].front().simulationTime == entity->getProperty<float>("m_flSimulationTime")))
+                continue;
+
+            while (records[i].size() > 3 && records[i].size() > (size_t)timeToTicks((float)config.backtrack.timeLimit / 1000.f))
+                records[i].pop_back();
+
+            auto varmap = reinterpret_cast<uintptr_t>(entity) + 0x24;
+            auto varscount = *reinterpret_cast<int*>((uintptr_t)(varmap) + 0x14);
+            for (int j = 0; j < varscount; j++)
+                *reinterpret_cast<uintptr_t*>(*reinterpret_cast<uintptr_t*>(varmap) + j * 0xC) = 0;
+
+            Record record{ };
+            record.absOrigin = entity->getAbsOrigin();
+            record.mins = entity->getProperty<Vector>("m_vecMins");
+            record.maxs = entity->getProperty<Vector>("m_vecMaxs");
+            record.head = entity->getBonePosition(8);
+            record.flags = entity->getProperty<int>("m_fFlags");
+            record.simulationTime = entity->getProperty<float>("m_flSimulationTime");
+
+            entity->setupBones(record.matrix, 128, 0x7FF00, memory.globalVars->currenttime);
+
+            for (int i = 0; i < getEntityAnimationLayerCount(entity); i++) {
+                auto layer = getEntityAnimationLayer(entity, i);
+                if (!layer)
+                    continue;
+
+                record.layerOrder[i] = layer->order;
+                record.layerSequence[i] = layer->sequence;
+                record.layerWeight[i] = layer->weight;
+                record.layerCycle[i] = layer->cycle;
+            }
+
+            records[i].emplace_front(record);
+        }
+    }
+}
+
+void Backtrack::run(UserCmd * cmd) noexcept
+{
+    if (!config.backtrack.enabled)
+        return;
+
+    if (!(cmd->buttons & UserCmd::IN_ATTACK))
+        return;
+
+    const auto localPlayer = interfaces.entityList->getEntity(interfaces.engine->getLocalPlayer());
+    if (!localPlayer)
+        return;
+
+    auto localPlayerEyePosition = localPlayer->getEyePosition();
+
+    auto bestFov{ std::numeric_limits<float>::infinity() };
```

Comment:
No need to use `std::numeric_limits<float>::infinity()` here - max fov in CS:GO is √(178^2 + 180^2) = 253.1481. That's why max fov in aimbot is 255 (rounded up).